src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellRenderer.cs

```diff
@@ -245,7 +244,6 @@ void OnDestroyed(object sender, EventArgs args)
 			{
 				previousView.Destroyed -= OnDestroyed;
 
-				previousView.Dispose();
```

Comment:
I'm thinking

1) make destroy internal but call it disconnect
2) call disconnect here instead of dispose
3) inside dispose still call disconnect

We should still keep that path open but not trigger it ourselves

Reply:
Chatted with @pictos on discord

This is fine
The destroy code is already called from inside "ShellItemRenderer"